documentation/SA1128.md

```diff
@@ -0,0 +1,90 @@
+## SA1128
+
+<table>
+<tr>
+  <td>TypeName</td>
+  <td>SA1128ConstructorInitializerMustBeOnOwnLine</td>
+</tr>
+<tr>
+  <td>CheckId</td>
+  <td>SA1128</td>
+</tr>
+<tr>
+  <td>Category</td>
+  <td>Readability Rules</td>
+</tr>
+</table>
+
+## Cause
+
+A constructor initializer is on the same line as the constructor declaration, within a C# code file.
+
+## Rule description
+
+A violation of this rule occurs whenever the code contains a constructor initializer that is partially or completely on the same line as the constructor declaration.
+
+For example, the following code would produce a violation of this rule:
+
+```csharp
+public class TypeName
+{
+    public TypeName() : this(0)
+    {
+    }
+
+    public TypeName(int value)
+    {
+    }
+}
+```
+
+The `:` (colon) character is treated as part of the initializer and should be on the same line as the `base` or `this` keyword.
+
+The following code would also produce a violation of this rule:
+
+```csharp
+public class TypeName
+{
+	public TypeName() :
+		base()
+	{
+	}
+}
+```
+
+The following code would not produce any violations:
+
+```csharp
+public class TypeName
+{
+    public TypeName() :
+        this(0)
+    {
+    }
+
+    public TypeName(int value)
+    {
+    }
+}
+```
+
+## How to fix violations
+
+To fix a violation of this rule, the constructor initializer, including the `:` character, to its own line.
+
+## How to suppress violations
+
+```csharp
+#pragma warning disable SA1128 // Constructor initializers must be on their own line
```

Comment:
:bulb: These directives can be placed around just the line containing the `:` (fixed during the merge).